File: domain/src/main/java/by/grodno/vasili/domain/repository/UserRepository.java
package by.grodno.vasili.domain.repository;

import java.util.List;

import by.grodno.vasili.domain.model.Organization;
import by.grodno.vasili.domain.model.User;
import io.reactivex.Maybe;
import io.reactivex.Single;

/**
 * Implement this interface for work with User
 */
public interface UserRepository {
    /**
     * Takes one {@link User} from datasource by id
     * @param id {@link User} identifier
     * @return Single observable {@link User}
     */
    Maybe<User> getOne(String id);

    /**
     * Takes all {@link User}`s from datasource
     * @param since The integer ID of the last User that you've seen
     * @return list of {@link User}
     */
    Single<List<User>> getAll(int since);

    /**
     * Take organizations for user from datasource
     *
     * @param username Username of user
     * @return Single observable list of {@link Organization}`s
     */
    Single<List<Organization>> getOrganizations(String username);
}

File: domain/src/main/java/by/grodno/vasili/domain/repository/UserRepository.kt
package by.grodno.vasili.domain.repository

import by.grodno.vasili.domain.model.Organization
import by.grodno.vasili.domain.model.User

/**
 * Implement this interface for work with User
 */
interface UserRepository {
    /**
     * Takes one [User] from datasource by id.
     * [id] [User] identifier.
     */
    suspend fun getOne(id: String): User

    /**
     * Takes all [User]`s from datasource.
     * [since] The integer ID of the last User that you've seen.
     */
    suspend fun getAll(since: Int): List<User>

    /**
     * Take organizations for user from datasource.
     *
     * [username] Username of user.
     */
    suspend fun getOrganizations(username: String): List<Organization>
}
